contracts/contribute/ContributeManager.sol

```diff
@@ -176,7 +176,7 @@ contract ContributeManager is ERC721TokenReceiver {
         instances[0] = msg.sender;
 
         // Create a service owned by this contract
-        serviceId = IService(serviceManager).create(address(this), token, configHash, agentIds,
+        serviceId = IService(serviceManager).create(contributorsProxy, token, configHash, agentIds,
```

Comment:
NO asset should ever be held in the periphery, that's the whole point of the design paradigm. You need to proxy everything via the core (which has a proxy design and can therefore be updated in case of need)